### DISCUSSION TITLE
Any success with GlowTTS -> Torchscript?

### DISCUSSION THREAD

**Discussion by ihshareef** [coqui-ai/TTS#491] [General]
The GlowTTS implementation is excellent. Great work.
I currently have a Glow-TTS model that I trained on `coqui-ai/TTS`, and I'm looking to utilise it on mobile. However, I've been quite unsuccessful at converting the model to either `TorchScript` or `ONNX`. Has anyone had any such success? If so, would they mind helping me out? 

Thanks

**Comment by nicolalandro** [comment]
I do not know if there is a method directly from this project, but by bring out the class of the pythorch model in a custom codebase (you only need the load model and predict) you can youse [pytorch mobile](https://pytorch.org/mobile/home/) to create application with Android or iOs or serve a coqi-ai/TTS server and use the api.

(I'm newest, sorry in advance if the answare is not so good)

**Comment by erogol** [comment]
We do not have ONNX  yet,  but in our plans. I know @synesthesiam is also interested in discovering more about it.

**Comment by synesthesiam** [comment]
As @erogol mentioned, I plan to look at ONNX export. I have it [already working elsewhere](https://github.com/rhasspy/glow-tts-train/blob/master/glow_tts_train/export_onnx.py), and I suspect the process will be straightforward since both our GlowTTS implementations come from the same source.

The only "trick" I needed to figure out was storing the inverse with `torch.no_grad`:

```python
# Inference only
model.eval()

# Do not calcuate jacobians for fast decoding
with torch.no_grad():
    model.decoder.store_inverse()
```

**Reply by AIFahim** [reply]
@synesthesiam Have you implemented ONNX Export till now?

**Reply by synesthesiam** [reply]
No, I've moved to working on Piper (https://github.com/rhasspy/piper) which uses VITS and is targeted at low end machines like the RPi 4.
